wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/ChePlugin.java

```diff
@@ -51,7 +53,13 @@ public ChePlugin id(String id) {
   }
 
   public String getId() {
-    return id;
+    if (!isNullOrEmpty(id)) {
+      return id;
+    } else if (!isNullOrEmpty(publisher) && !isNullOrEmpty(name) && !isNullOrEmpty(version)) {
+      return String.format("%s/%s/%s", publisher, name, version);
```

Comment:
Are you sure that ```String.format``` is one of the effective ways to concatenate three strings together?

Reply:
I'm sure it is not. fixed